addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/pscan/WebSocketPassiveScannerManager.java

```diff
@@ -0,0 +1,244 @@
+     * @param tableWebSocket the table
+     */
+    public void setTableWebSocketIfNot(TableWebSocket tableWebSocket){
+        if(!getWebSocketPassiveScanThread().hasTableWebSocket()){
+            passiveScanThread.setTableWebSocket(tableWebSocket);
+        }
+    }
+    
+    /**
+     * Setting the table WebSocket
+     */
+    public void setTableWebSocket(TableWebSocket tableWebSocket){
+        passiveScanThread.setTableWebSocket(tableWebSocket);
+    }
+    
+    /**
+     * Adds the WebSocketPassive Scanner if not null
+     */
+    synchronized public WebSocketPassiveScannerPlugin addPassiveScannerPlugin(WebSocketPassiveScanner passiveScanner) {
+        if (passiveScanner == null) {
+            throw new IllegalArgumentException("Parameter passiveScanner must not be null.");
+        }
+        WebSocketPassiveScannerPlugin wsPlugin = new WebSocketPassiveScannerPlugin(passiveScanner);
+        return add(new WebSocketPassiveScannerPlugin(passiveScanner)) ? wsPlugin : null;
+    }
+    
+    /**
+     * Enables or disables all WebSocket Passive Scanners
+     * @param enabled {@code true} if the scanners should be enabled, {@code false} otherwise
+     */
+    public void setAllPluginPassiveScannersEnabled(boolean enabled) {
+        Iterator<WebSocketPassiveScanner> iterator = getIterator();
+        WebSocketPassiveScannerPlugin scanner;
+        while (iterator.hasNext()) {
+            scanner = (WebSocketPassiveScannerPlugin) iterator.next();
+            scanner.setEnabled(enabled);
+        }
+    }
+    
+    /**
+     * Shut down the background thread if any.
+     */
+    private void stopWebSocketPassiveScanThread() {
+        if (this.passiveScanThread != null) {
+            passiveScanThread.shutdown();
+        }
+    }
+    
+    /**
+     * Activating or Deactivating the passive scanning.
+     * That's not related with any {@link WebSocketPassiveScannerPlugin}
+     * just with background thread {@link WebSocketPassiveScanThread}
+     * @param enabled if true activates the background thread
+     */
+    public void setPassiveScanEnabled(boolean enabled) {
+        if (passiveScanEnabled != enabled) {
+            passiveScanEnabled = enabled;
+            if (enabled) {
+                startWebSocketPassiveScanThread();
+            } else {
+                stopWebSocketPassiveScanThread();
+            }
+        }
+    }
+    
+    public AlertManager getAlertManager() {
+        return alertManager;
+    }
+    
+    public void setAlertManager(AlertManager alertManager) {
+        this.alertManager = alertManager;
+    }
+    
+    /**
+     * Add a passive scanner to thread safe list
+     * @return {@code true} if passive scanner is added to list successfully.
+     */
+    private boolean add(WebSocketPassiveScanner passiveScanner){
+        if(getPassiveScannersSet().contains(passiveScanner)){
+            LOGGER.warn("Insertion prevent in order to avoid the duplication");
```

Comment:
Should this include some details like the scanner name that collided or something?